core/src/Cabana_GaussianMixtureModel.hpp

```diff
@@ -0,0 +1,57 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the Cabana authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+/*!
+  \file Cabana_GaussianMixtureModel.hpp
+  \brief Creation of a Gaussian Mixture Model
+*/
+#ifndef CABANA_GMM_HPP
+#define CABANA_GMM_HPP
+
+#include <typeinfo>
+
+// Things to describe a Gaussian
+enum GaussianFields {
```

Comment:
This works, but requires a user to have exactly this data in their AoSoA. We have new capability for naming and accessing fields with type tags (Cabana_Fields.hpp, used in Cabana_ParticleList.hpp, which ultimately just wraps an AoSoA)

Reply:
That is something the user GETS from us, so I assume it's reasonable to assume fixed names.